ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs

```diff
@@ -45,9 +48,128 @@ import           Ouroboros.Consensus.HardFork.Combinator.Ledger.Query
 import           Ouroboros.Consensus.HardFork.Combinator.Mempool
 import           Ouroboros.Consensus.HardFork.Combinator.Serialisation.Common
 import           Ouroboros.Consensus.HardFork.Combinator.Serialisation.SerialiseDisk ()
+import qualified Ouroboros.Consensus.HardFork.History as History
+
+{-------------------------------------------------------------------------------
+  Serialisation of products
+-------------------------------------------------------------------------------}
+
+-- | Encoding of `NP f xs` while filtering out the components for disabled eras
+-- as indicated by the `HardForkNodeToClientVersion xs`. Disabled eras imply
+-- that the protocol version does not support those eras. Hence, omitting the
+-- corresponding elements is the correct behavior.
+encodeNodeToClientNP ::
```

Comment:
I think we should make `f` the first type variable, not `xs`.

Also, style: the first line layout should be

```
encodeNodeToClientNP :: forall f xs. SerialiseHFC xs
  => ...
```

Reply:
Doesn't the style guide suggest that we put the forall on a new line:

```haskell
fooBar ::
     forall a .. z. (Eq a, ..)
  => a
  -> ..
  -> ..
```

Reply:
Ah, egg on my face.

Yeah, follow the guide. Thanks for catching my mis-remembering.